Druid/DRCogitStackToRegisterMappingGenerator.class.st

```diff
@@ -256,17 +258,24 @@ DRCogitStackToRegisterMappingGenerator >> visitContinueNextBytecode: aDRContinue
 		target: #nextBytecode
 ]
 
+{ #category : #visiting }
+DRCogitStackToRegisterMappingGenerator >> visitFlushStack: aDRFlushStack [
+
+	generatorMethodBuilder addStatement: (RBMessageNode
+			 receiver: RBVariableNode selfNode
+			 selector: #ssFlushStack)
+]
+
 { #category : #visiting }
 DRCogitStackToRegisterMappingGenerator >> visitJITMessageSend: aDRMessageSend [
 
 	"self genMarshalledSend: selectorIndex numArgs: numArgs sendTable: ordinarySendTrampolines"
-
 	generatorMethodBuilder addStatement: (RBMessageNode
 			 receiver: RBVariableNode selfNode
 			 selector: #genMarshalledSend:numArgs:sendTable:
-			 arguments: { 
-				RBLiteralValueNode value: aDRMessageSend cogitSelectorIndex.
-				RBLiteralValueNode value: aDRMessageSend numberOfArguments.
+			 arguments: {
+				(aDRMessageSend operand1 rtlPushArgumentExpressions: self) first.
+				(aDRMessageSend operand2 rtlPushArgumentExpressions: self) first.
```

Comment:
🤔

Reply:
Here also this was the case where selector index and number of arguments were constants. 
Now they could be computed (and in the case they are constants they will be constant propagated), and be modelled as normal instructions. So we need to visit them.